sfputil/main.py

```diff
@@ -1112,6 +1112,64 @@ def run_firmware(port_name, mode):
 
     return status
 
+def is_fw_switch_done(port_name):
+    """
+        Make sure the run_firmware cmd is done
+        @port_name:
+        Returns 1 on success, and exit_code = -1 on failure
+    """
+    status = 0
+    physical_port = logical_port_to_physical_port_index(port_name)
+    sfp = platform_chassis.get_sfp(physical_port)
+
+    try:
+        api = sfp.get_xcvr_api()
+    except NotImplementedError:
+        click.echo("This functionality is currently not implemented for this platform")
+        sys.exit(ERROR_NOT_IMPLEMENTED)
+
+    try:
+        MAX_WAIT = 600 # 60s timeout.
```

Comment:
can we read from the module what is advertised instead of hardcoding here?

Reply:
There is no definition of max duration for CDB_run cmd until CMIS5.2.
![CDB_Duration](https://user-images.githubusercontent.com/28720623/193563116-6169469c-0330-40e8-8a33-3cd65c72b28b.png)

Reply:
@CliveNi why 60 secs?

Reply:
For more compatible with different module, timeout time should be the worst case. 60s is calculated for different implementation.
Feasible implementation as below, 
1. Dual image(A and B) is supported, and fw only executable at specific image.
2. Dual image(A and B) is supported, and fw executable at both image.
3. Triple image(A, B and factory) is supported, and fw only executable at specific image.
3. Triple image(A, B and factory) is supported, and fw executable at all image.
4. Single image is supported.

Case 1 and 3 will spend the most time to switch image. It will need to copy one image to non-volatile buffer, then interchange two images with each other. That will need 3 cycles erasing and programming on non-volatile memory.

We can roughly assume all of fw component in module around 1MBytes. A common M0~M4 MCU needs around 40ms per page(2KBytes) for non-volatile erasing and programming, so that the calculation as below.
1MBytes / 2KBytes * 40ms * 3cycles = 61.44s

Reply:
Just curious, if a firmware size is >> 1MB (e.g. close to 2 MB), does that mean 60 sec  wait time would not be enough?

Reply:
Exactly, 1MB is a roughly number, but I think the margin is enough for different DSP solution now. Actually, it only uses around 500KB totally in our 400/800G implementation.